cupy_backends/cuda/cupy_cusparse.h

```diff
@@ -5,201 +5,440 @@
+ * versions of CUDA, so we have to provide a stub implementation using the full signature
+ * declared in the header, instead of using (...).
+ */
+
+#if defined(_WIN32) && (CUSPARSE_VERSION < 11000)
+  #if 10200 < CUSPARSE_VERSION
+    #define WIN32_EXPOSE_SPMM_STUB_DECL 1
+  #else
+    #define WIN32_EXPOSE_SPMM_STUB_DECL 0
+  #endif
+  #define WIN32_EXPOSE_SPMM_STUB_IMPL 1
 #else
-#define CUSPARSE_VERSION 10000
+  #define WIN32_EXPOSE_SPMM_STUB_DECL 0
+  #define WIN32_EXPOSE_SPMM_STUB_IMPL 0
 #endif
-#endif // #if !defined(CUSPARSE_VERSION)
 
-#if (CUSPARSE_VERSION < 10200) || (CUSPARSE_VERSION < 11000 && defined(_WIN32))
-// Types, macro and functions added in cuSparse 10.2
-// Windows support added in cuSparse 11.0
+#if CUSPARSE_VERSION < 10010
+// Added in cuSPARSE 10.1 (CUDA 10.1.105)
+
+// CSR2CSC
+typedef enum {} cusparseCsr2CscAlg_t;
+
+#endif  // #if CUSPARSE_VERSION < 10010
+
+#if CUSPARSE_VERSION < 10010 || WIN32_EXPOSE_SPMM_STUB_DECL
+// Generic API types added in cuSPARSE 10.1 (CUDA 10.1.105)
 
-// cuSPARSE generic API
-typedef void* cusparseSpVecDescr_t;
-typedef void* cusparseDnVecDescr_t;
 typedef void* cusparseSpMatDescr_t;
 typedef void* cusparseDnMatDescr_t;
-
 typedef enum {} cusparseIndexType_t;
 typedef enum {} cusparseFormat_t;
 typedef enum {} cusparseOrder_t;
-typedef enum {} cusparseSpMVAlg_t;
 typedef enum {} cusparseSpMMAlg_t;
 
-cusparseStatus_t cusparseCreateSpVec(...) {
+#endif  // #if CUSPARSE_VERSION < 10010 || WIN32_EXPOSE_SPMM_STUB_DECL
+
+#if CUSPARSE_VERSION < 10200
+// Added in cuSPARSE 10.2 (CUDA 10.1.168)
+
+// CSR2CSC
+cusparseStatus_t cusparseCsr2cscEx2_bufferSize(...) {
   return CUSPARSE_STATUS_SUCCESS;
 }
 
-cusparseStatus_t cusparseDestroySpVec(...) {
+cusparseStatus_t cusparseCsr2cscEx2(...) {
   return CUSPARSE_STATUS_SUCCESS;
 }
 
-cusparseStatus_t cusparseSpVecGet(...) {
+#endif  // #if CUSPARSE_VERSION < 10200
+
+#if CUSPARSE_VERSION < 10200 || WIN32_EXPOSE_SPMM_STUB_DECL
+// Generic API types added in cuSPARSE 10.2 (CUDA 10.1.168)
+
+typedef void* cusparseSpVecDescr_t;
+typedef void* cusparseDnVecDescr_t;
+typedef enum {} cusparseSpMVAlg_t;
+
+#endif  // #if CUSPARSE_VERSION < 10200 || WIN32_EXPOSE_SPMM_STUB_DECL
+
+#if CUSPARSE_VERSION < 10200 || WIN32_EXPOSE_SPMM_STUB_IMPL
+// Generic APIs added in cuSPARSE 10.2 (CUDA 10.1.168)
+
+cusparseStatus_t cusparseCreateSpVec(
+    cusparseSpVecDescr_t* spVecDescr,
+    int64_t               size,
+    int64_t               nnz,
+    void*                 indices,
+    void*                 values,
+    cusparseIndexType_t   idxType,
+    cusparseIndexBase_t   idxBase,
+    cudaDataType          valueType) {
   return CUSPARSE_STATUS_SUCCESS;
 }
 
-cusparseStatus_t cusparseSpVecGetIndexBase(...) {
+cusparseStatus_t cusparseDestroySpVec(cusparseSpVecDescr_t spVecDescr) {
   return CUSPARSE_STATUS_SUCCESS;
 }
 
-cusparseStatus_t cusparseSpVecGetValues(...) {
+cusparseStatus_t cusparseSpVecGet(
```

Comment:
nit: would be useful to have a comment here saying that the args were added to fix an issue in windows.

Reply:
just a ref to the PR should be enough

Reply:
Thanks! Updated comments.